pydicom/charset.py

```diff
@@ -122,23 +122,34 @@ def _encode_to_jis_x_0201(value, errors='strict'):
     ------
     UnicodeEncodeError
         If errors is set to 'strict' and `value` could not be encoded with
-        JIX X 0201.
+        JIS X 0201.
     """
 
-    # If errors is not strict, this function is used in fallback.
-    # So keep the tail escape sequence of encoded for backward compatibility.
-    if errors != 'strict' or value == '':
-        return value.encode('shift_jis', errors=errors)
-
     Encoder = codecs.getincrementalencoder('shift_jis')
     encoder = Encoder()
 
+    # If errors is not strict, this function is used as fallback.
+    # In this case, we use only ISO IR 14 to encode given value
+    # without escape sequence.
+    if errors != 'strict' or value == '':
+        encoded = b''
+        for c in value:
+            try:
+                b = encoder.encode(c)
+            except UnicodeEncodeError as e:
+                b = b'?'
```

Comment:
One thing I missed last time: we usually issue a warning if the encoding fails and we are not in `strict` mode (as is done in `encode_string`, for example). Could you add this please, and adapt the test to also check for the warning (e.g. using `with pytest.warns()`)?

Reply:
Thanks - looks like I misunderstood the code again :frowning:, but a test in place always helps...

Reply:
Thanks for your quick reply! I got it!